src/services/dumpers/agent-nodejs.ts

```diff
@@ -119,10 +157,18 @@ export default class AgentNodeJs extends AbstractDumper {
     };
 
     if (isMongoose) {
-      context.datasourceImport = `const { createMongooseDataSource } = require('@forestadmin/datasource-mongoose');\nconst connection = require('./models');`;
+      if (language === languages.Typescript) {
+        context.datasourceImport = `import { createMongooseDataSource } from '@forestadmin/datasource-mongoose';\nimport connection from './models';`;
+      } else {
+        context.datasourceImport = `const { createMongooseDataSource } = require('@forestadmin/datasource-mongoose');\nconst connection = require('./models');`;
+      }
       context.datasourceCreation = `createMongooseDataSource(connection, { flattenMode: 'auto' })`;
     } else {
-      context.datasourceImport = `const { createSqlDataSource } = require('@forestadmin/datasource-sql');`;
+      if (language === languages.Typescript) {
+        context.datasourceImport = `import { createSqlDataSource } from '@forestadmin/datasource-sql';`;
+      } else {
+        context.datasourceImport = `const { createSqlDataSource } = require('@forestadmin/datasource-sql');`;
+      }
```

Comment:
I don't get why this isn't a part of the template, where you already already know the langage ?

Reply:
@SteveBunlon